src/Components/DropdownInput/DropdownInput.tsx

```diff
@@ -0,0 +1,42 @@
+import debounce from 'lodash.debounce';
+import { useCallback } from 'react';
+
+type Props = {
+  delay: number;
+  query: string;
+  setQuery: (query: string) => void;
+  setAppliedQuery: (query: string) => void;
+  handleOnBlur: () => void;
+  handleOnFocus: () => void;
+};
+
+export const DropdownInput: React.FC<Props> = ({
+  delay,
+  query,
+  setQuery,
+  setAppliedQuery,
+  handleOnBlur,
+  handleOnFocus,
+}) => {
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const applyQuery = useCallback(debounce(setAppliedQuery, delay), []);
+
+  const handleOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    applyQuery(event.target.value);
+    setQuery(event.target.value);
+  };
```

Comment:
I would rather move these `handleOnChange`  and `applyQuery ` up to the `Dropdown` component and push like props